relay_messages.go

```diff
@@ -113,26 +140,7 @@ func (f lazyCallReq) Service() string {
 // Method returns the name of the method being called. It panics if called for
 // a non-callReq frame.
 func (f lazyCallReq) Method() string {
-	serviceLen := f.Payload[_serviceLenIndex]
-
-	// nh:1 (hk~1 hv~1){nh}
-	headerStart := _serviceLenIndex + 1 /* length byte */ + serviceLen
-	numHeaders := int(f.Payload[headerStart])
-	cur := int(headerStart) + 1
-	for i := 0; i < numHeaders*2; i++ {
-		sLen := f.Payload[cur]
-		cur += 1 + int(sLen)
-	}
-
-	// csumtype:1 (csum:4){0,1} arg1~2 arg2~2 arg3~2
-	checkSumType := ChecksumType(f.Payload[cur])
-	cur += 1 /* checksum */ + checkSumType.ChecksumSize()
-
-	// arg1~2
-	arg1Len := int(binary.BigEndian.Uint16(f.Payload[cur : cur+2]))
-	cur += 2
-	arg1 := f.Payload[cur : cur+arg1Len]
-	return string(arg1)
+	return string(f.method)
```

Comment:
true, made `Caller` and `Method` use a string backed private field. I was initially against it since the performance is different for `Caller` vs `Service`, but better to have it be faster now, and make it all consistent when we move everything to `[]byte`.

While we could also add a private `string` field for `Service`, I'd like to avoid adding private fields for anything that is at a fixed offset in the payload. If we keep `lazyCallReq` very small, we can avoid pooling and just pass the value around and avoid some more complication in the code.